packages/odyssey-react-mui/src/Autocomplete.tsx

```diff
@@ -371,60 +396,95 @@ const Autocomplete = <
     return <div ref={ref} {...props} {...outerProps} />;
   });
 
-  function useResetCache(length: number) {
+  const useResetCache = (length: number) => {
     const ref = useRef<VariableSizeList>(null);
     useEffect(() => {
       if (ref.current) {
         ref.current.resetAfterIndex(0, true);
       }
     }, [length]);
     return ref;
-  }
+  };
```

Comment:
*NON-BLOCKING* We should probably rename this `ref` to `resetCacheRef` internally, so it's more clear. Not required to make this change since it was already approved.

I think we have manipulation logic for this ref somewhere else right?

If we're doing it elsewhere, we should probably move it into this hook instead. In general, when you create a custom hook, think of it like a `class`. It should include everything it needs including how to manipulate it.

You can return functions in the return statement:
```ts
{
  resetCacheRef,
  doSomeManipulation,
  doAnotherManipulation,
}
```